magefile.go

```diff
@@ -18,7 +19,8 @@ type Test mg.Namespace
 func (Test) Run() error {
 	goCmd := mg.GoCmd()
 	env := map[string]string{
-		"APP_ENV": "test",
+		"APP_ENV":    "test",
+		"CGO_CFLAGS": "-g -O2 -Wno-return-local-addr",
```

Comment:
@laynax Sure thing, here we go:

```bash
Creating go-sdk_sdk_run ... done
Running target: Test:Run
exec: go test ./... -count=1 -v
=== RUN   TestViperBuilder
=== RUN   TestViperBuilder/Build
=== RUN   TestViperBuilder/Build/ValidFile
=== RUN   TestViperBuilder/Build/ValidWithoutEnvs
=== RUN   TestViperBuilder/Build/InvalidFile
=== RUN   TestViperBuilder/Build/NonExistentFile
--- PASS: TestViperBuilder (0.06s)
    --- PASS: TestViperBuilder/Build (0.06s)
        --- PASS: TestViperBuilder/Build/ValidFile (0.01s)
        --- PASS: TestViperBuilder/Build/ValidWithoutEnvs (0.01s)
        --- PASS: TestViperBuilder/Build/InvalidFile (0.01s)
        --- PASS: TestViperBuilder/Build/NonExistentFile (0.02s)
PASS
ok      github.com/scribd/go-sdk/internal/pkg/configuration/builder     0.088s
=== RUN   TestNewConfig
=== RUN   TestNewConfig/ValidFile
=== RUN   TestNewConfig/InvalidFile
=== RUN   TestNewConfig/NonExistentFile
--- PASS: TestNewConfig (0.05s)
    --- PASS: TestNewConfig/ValidFile (0.01s)
    --- PASS: TestNewConfig/InvalidFile (0.02s)
    --- PASS: TestNewConfig/NonExistentFile (0.02s)
=== RUN   TestGetters
--- PASS: TestGetters (0.02s)
PASS
ok      github.com/scribd/go-sdk/pkg/app        0.089s
?       github.com/scribd/go-sdk/pkg/configuration      [no test files]
?       github.com/scribd/go-sdk/pkg/context/database   [no test files]
?       github.com/scribd/go-sdk/pkg/context/logger     [no test files]
?       github.com/scribd/go-sdk/pkg/context/metrics    [no test files]
=== RUN   TestExtract
=== RUN   TestExtract/Context_without_request_id
=== RUN   TestExtract/Context_contains_request_id
--- PASS: TestExtract (0.00s)
    --- PASS: TestExtract/Context_without_request_id (0.00s)
    --- PASS: TestExtract/Context_contains_request_id (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/context/requestid  0.040s
?       github.com/scribd/go-sdk/pkg/contextkeys        [no test files]
=== RUN   TestNewConfig
=== RUN   TestNewConfig/RunningInTestEnvironment
=== RUN   TestNewConfig/NewWithoutConfigFileFails
--- PASS: TestNewConfig (0.02s)
    --- PASS: TestNewConfig/RunningInTestEnvironment (0.00s)
    --- PASS: TestNewConfig/NewWithoutConfigFileFails (0.02s)
=== RUN   TestNewConnectionDetails
    connection_details_test.go:12: Cannot create NewConfig: Config File "database" Not Found in "[/sdk/pkg/database/config]"
--- PASS: TestNewConnectionDetails (0.01s)
=== RUN   TestString
=== RUN   TestString/WithAllAttributesPresent
=== RUN   TestString/WithOneAttributeBlank
--- PASS: TestString (0.00s)
    --- PASS: TestString/WithAllAttributesPresent (0.00s)
    --- PASS: TestString/WithOneAttributeBlank (0.00s)
=== RUN   TestStringWithoutDB
=== RUN   TestStringWithoutDB/WithAllAttributesPresent
=== RUN   TestStringWithoutDB/WithOneAttributeBlank
--- PASS: TestStringWithoutDB (0.00s)
    --- PASS: TestStringWithoutDB/WithAllAttributesPresent (0.00s)
    --- PASS: TestStringWithoutDB/WithOneAttributeBlank (0.00s)
=== RUN   TestOpts
=== RUN   TestOpts/WithPresentTimeout
=== RUN   TestOpts/WithBlankTimeout
--- PASS: TestOpts (0.00s)
    --- PASS: TestOpts/WithPresentTimeout (0.00s)
    --- PASS: TestOpts/WithBlankTimeout (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/database   0.094s
=== RUN   TestInstrumentAWSSession
=== RUN   TestInstrumentAWSSession/s3
--- PASS: TestInstrumentAWSSession (0.65s)
    --- PASS: TestInstrumentAWSSession/s3 (0.65s)
=== RUN   TestNewConfig
=== RUN   TestNewConfig/NewWithoutConfigFileFails
--- PASS: TestNewConfig (0.01s)
    --- PASS: TestNewConfig/NewWithoutConfigFileFails (0.01s)
=== RUN   TestNewConfigWithAppRoot
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks
--- PASS: TestNewConfigWithAppRoot (0.00s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks (0.00s)
=== RUN   TestNewConfigWithAppRootAndOverwriteFromEnvTheEnableFlag
=== RUN   TestNewConfigWithAppRootAndOverwriteFromEnvTheEnableFlag/NewWithConfigWithEnvVariablesOverwritten
--- PASS: TestNewConfigWithAppRootAndOverwriteFromEnvTheEnableFlag (0.01s)
    --- PASS: TestNewConfigWithAppRootAndOverwriteFromEnvTheEnableFlag/NewWithConfigWithEnvVariablesOverwritten (0.01s)
=== RUN   TestInstrumentDatabase
--- PASS: TestInstrumentDatabase (0.01s)
=== RUN   TestTraceLogs
--- PASS: TestTraceLogs (0.00s)
=== RUN   TestProfilerStop
--- PASS: TestProfilerStop (0.00s)
=== RUN   TestNewTracer
=== RUN   TestNewTracer/WithAConfigurationIsAlwaysOkWhenEnabled
=== RUN   TestNewTracer/WithAConfigurationIsAlwaysOkWhenDisabled
--- PASS: TestNewTracer (0.00s)
    --- PASS: TestNewTracer/WithAConfigurationIsAlwaysOkWhenEnabled (0.00s)
    --- PASS: TestNewTracer/WithAConfigurationIsAlwaysOkWhenDisabled (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/instrumentation    0.688s
=== RUN   TestNewClient
--- PASS: TestNewClient (0.00s)
=== RUN   TestAnalyticsSettings
=== RUN   TestAnalyticsSettings/defaults
=== RUN   TestAnalyticsSettings/enabled
=== RUN   TestAnalyticsSettings/provide_analytics_rate
=== RUN   TestAnalyticsSettings/provide_analytics_rate/less_than_0.0
=== RUN   TestAnalyticsSettings/provide_analytics_rate/more_than_1.0
=== RUN   TestAnalyticsSettings/provide_analytics_rate/in_allowed_range
--- PASS: TestAnalyticsSettings (0.00s)
    --- PASS: TestAnalyticsSettings/defaults (0.00s)
    --- PASS: TestAnalyticsSettings/enabled (0.00s)
    --- PASS: TestAnalyticsSettings/provide_analytics_rate (0.00s)
        --- PASS: TestAnalyticsSettings/provide_analytics_rate/less_than_0.0 (0.00s)
        --- PASS: TestAnalyticsSettings/provide_analytics_rate/more_than_1.0 (0.00s)
        --- PASS: TestAnalyticsSettings/provide_analytics_rate/in_allowed_range (0.00s)
=== RUN   TestWithServiceName
=== RUN   TestWithServiceName/defaults
=== RUN   TestWithServiceName/service_name_provided
--- PASS: TestWithServiceName (0.00s)
    --- PASS: TestWithServiceName/defaults (0.00s)
    --- PASS: TestWithServiceName/service_name_provided (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/instrumentation/kafka      0.036s
=== RUN   TestDatabaseLoggingUnaryServerInterceptor
    database_logging_test.go:57: Errors: table "test_records" already exists

(/sdk/pkg/interceptors/database_logging_test.go:51) 
[2022-06-29 13:05:01]  table "test_records" already exists 
--- FAIL: TestDatabaseLoggingUnaryServerInterceptor (0.08s)
=== RUN   TestDatabaseLoggingStreamServerInterceptors
--- PASS: TestDatabaseLoggingStreamServerInterceptors (0.01s)
=== RUN   TestLoggerUnaryServerInterceptors
--- PASS: TestLoggerUnaryServerInterceptors (0.00s)
=== RUN   TestLoggerStreamServerInterceptors
--- PASS: TestLoggerStreamServerInterceptors (0.00s)
=== RUN   TestRequestIDUnaryServerInterceptor
=== RUN   TestRequestIDUnaryServerInterceptor/without_request_ID
=== RUN   TestRequestIDUnaryServerInterceptor/with_request_ID
--- PASS: TestRequestIDUnaryServerInterceptor (0.00s)
    --- PASS: TestRequestIDUnaryServerInterceptor/without_request_ID (0.00s)
    --- PASS: TestRequestIDUnaryServerInterceptor/with_request_ID (0.00s)
=== RUN   TestRequestIDStreamServerInterceptor
=== RUN   TestRequestIDStreamServerInterceptor/without_request_id
=== RUN   TestRequestIDStreamServerInterceptor/with_request_id
--- PASS: TestRequestIDStreamServerInterceptor (0.00s)
    --- PASS: TestRequestIDStreamServerInterceptor/without_request_id (0.00s)
    --- PASS: TestRequestIDStreamServerInterceptor/with_request_id (0.00s)
FAIL
FAIL    github.com/scribd/go-sdk/pkg/interceptors       0.106s
=== RUN   TestNewBuilder
=== RUN   TestNewBuilder/NewBuilderWithConfigFileSetValues
--- PASS: TestNewBuilder (0.00s)
    --- PASS: TestNewBuilder/NewBuilderWithConfigFileSetValues (0.00s)
=== RUN   TestSetFields
=== RUN   TestSetFields/SetFieldsWithValues
=== RUN   TestSetFields/SetFieldsWithoutValues
--- PASS: TestSetFields (0.00s)
    --- PASS: TestSetFields/SetFieldsWithValues (0.00s)
    --- PASS: TestSetFields/SetFieldsWithoutValues (0.00s)
=== RUN   TestSetTracking
=== RUN   TestSetTracking/SetTrackingIsTrueWithATrackingConfig
=== RUN   TestSetTracking/SetTrackingIsFalseWithoutATrackingConfig
--- PASS: TestSetTracking (0.00s)
    --- PASS: TestSetTracking/SetTrackingIsTrueWithATrackingConfig (0.00s)
    --- PASS: TestSetTracking/SetTrackingIsFalseWithoutATrackingConfig (0.00s)
=== RUN   TestBuild
=== RUN   TestBuild/WithATrackingConfigAndFieldsItBuilds
=== RUN   TestBuild/WithoutATrackingConfigAndEmptyFieldsItBuilds
--- PASS: TestBuild (0.88s)
    --- PASS: TestBuild/WithATrackingConfigAndFieldsItBuilds (0.88s)
    --- PASS: TestBuild/WithoutATrackingConfigAndEmptyFieldsItBuilds (0.00s)
=== RUN   TestBuildTestLogger
=== RUN   TestBuildTestLogger/WhenBuildingATestLoggerIsNotTracking
--- PASS: TestBuildTestLogger (0.00s)
    --- PASS: TestBuildTestLogger/WhenBuildingATestLoggerIsNotTracking (0.00s)
=== RUN   TestNewConfig
=== RUN   TestNewConfig/RunningInTestEnvironment
=== RUN   TestNewConfig/NewWithoutConfigFileFails
--- PASS: TestNewConfig (0.01s)
    --- PASS: TestNewConfig/RunningInTestEnvironment (0.00s)
    --- PASS: TestNewConfig/NewWithoutConfigFileFails (0.01s)
=== RUN   TestFields
=== RUN   TestFields/SetANewKey
=== RUN   TestFields/UpdatesAnExistingKey
--- PASS: TestFields (0.00s)
    --- PASS: TestFields/SetANewKey (0.00s)
    --- PASS: TestFields/UpdatesAnExistingKey (0.00s)
=== RUN   TestNewGormLogger
=== RUN   TestNewGormLogger/Empty_on_fatal_log_level
=== RUN   TestNewGormLogger/Empty_on_error_log_level
=== RUN   TestNewGormLogger/Empty_on_warning_log_level
=== RUN   TestNewGormLogger/Empty_on_info_log_level
=== RUN   TestNewGormLogger/Print_database_log
=== RUN   TestNewGormLogger/Empty_on_non_sql_input_in_debug_mode
=== RUN   TestNewGormLogger/Print_log_in_trace_mode_on_non_sql_input
--- PASS: TestNewGormLogger (0.00s)
    --- PASS: TestNewGormLogger/Empty_on_fatal_log_level (0.00s)
    --- PASS: TestNewGormLogger/Empty_on_error_log_level (0.00s)
    --- PASS: TestNewGormLogger/Empty_on_warning_log_level (0.00s)
    --- PASS: TestNewGormLogger/Empty_on_info_log_level (0.00s)
    --- PASS: TestNewGormLogger/Print_database_log (0.00s)
    --- PASS: TestNewGormLogger/Empty_on_non_sql_input_in_debug_mode (0.00s)
    --- PASS: TestNewGormLogger/Print_log_in_trace_mode_on_non_sql_input (0.00s)
=== RUN   TestInfoLevelWithJSONFields
--- PASS: TestInfoLevelWithJSONFields (0.00s)
=== RUN   TestInfoLevelWithTextFields
--- PASS: TestInfoLevelWithTextFields (0.00s)
=== RUN   TestLevelConfiguration
=== RUN   TestLevelConfiguration/RunningWithLoggerInTestEnvironment
=== RUN   TestLevelConfiguration/WhenConfigLevelIsTraceDebugIsLogged
=== RUN   TestLevelConfiguration/WhenConfigLevelIsWarnInfoIsNotLogged
--- PASS: TestLevelConfiguration (0.00s)
    --- PASS: TestLevelConfiguration/RunningWithLoggerInTestEnvironment (0.00s)
    --- PASS: TestLevelConfiguration/WhenConfigLevelIsTraceDebugIsLogged (0.00s)
    --- PASS: TestLevelConfiguration/WhenConfigLevelIsWarnInfoIsNotLogged (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/logger     0.944s
=== RUN   TestNewKafkaLogger
=== RUN   TestNewKafkaLogger/No_level_provided
=== RUN   TestNewKafkaLogger/Panic_level_provided
=== RUN   TestNewKafkaLogger/Fatal_level_provided
=== RUN   TestNewKafkaLogger/Error_level_provided
=== RUN   TestNewKafkaLogger/Warn_level_provided
=== RUN   TestNewKafkaLogger/Info_level_provided
=== RUN   TestNewKafkaLogger/Debug_level_provided
=== RUN   TestNewKafkaLogger/Trace_level_provided
=== RUN   TestNewKafkaLogger/WithLevelFn_provided
--- PASS: TestNewKafkaLogger (0.00s)
    --- PASS: TestNewKafkaLogger/No_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Panic_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Fatal_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Error_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Warn_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Info_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Debug_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/Trace_level_provided (0.00s)
    --- PASS: TestNewKafkaLogger/WithLevelFn_provided (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/logger/kafka       0.025s
?       github.com/scribd/go-sdk/pkg/metrics    [no test files]
=== RUN   TestNewBrokerMetrics
=== RUN   TestNewBrokerMetrics/no_sample_rate_specified
=== RUN   TestNewBrokerMetrics/sample_rate_specified
=== RUN   TestNewBrokerMetrics/sample_rate_specified_for_metric_name
--- PASS: TestNewBrokerMetrics (0.00s)
    --- PASS: TestNewBrokerMetrics/no_sample_rate_specified (0.00s)
    --- PASS: TestNewBrokerMetrics/sample_rate_specified (0.00s)
    --- PASS: TestNewBrokerMetrics/sample_rate_specified_for_metric_name (0.00s)
=== RUN   TestNewConsumerMetrics
=== RUN   TestNewConsumerMetrics/no_sample_rate_specified
=== RUN   TestNewConsumerMetrics/sample_rate_specified
=== RUN   TestNewConsumerMetrics/sample_rate_specified_for_metric_name
--- PASS: TestNewConsumerMetrics (0.00s)
    --- PASS: TestNewConsumerMetrics/no_sample_rate_specified (0.00s)
    --- PASS: TestNewConsumerMetrics/sample_rate_specified (0.00s)
    --- PASS: TestNewConsumerMetrics/sample_rate_specified_for_metric_name (0.00s)
=== RUN   TestNewProducerMetrics
=== RUN   TestNewProducerMetrics/no_sample_rate_specified
=== RUN   TestNewProducerMetrics/sample_rate_specified
=== RUN   TestNewProducerMetrics/sample_rate_specified_for_metric_name
--- PASS: TestNewProducerMetrics (0.00s)
    --- PASS: TestNewProducerMetrics/no_sample_rate_specified (0.00s)
    --- PASS: TestNewProducerMetrics/sample_rate_specified (0.00s)
    --- PASS: TestNewProducerMetrics/sample_rate_specified_for_metric_name (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/metrics/kafka      0.026s
=== RUN   TestNewDatabaseLoggingMiddleware
--- PASS: TestNewDatabaseLoggingMiddleware (0.11s)
=== RUN   TestOutputStructuredContentFromMiddleware
--- PASS: TestOutputStructuredContentFromMiddleware (0.00s)
=== RUN   TestResponseStatusFromMiddleware
--- PASS: TestResponseStatusFromMiddleware (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/middleware 0.120s
=== RUN   TestNewConfig
=== RUN   TestNewConfig/NewWithoutConfigFileFails
--- PASS: TestNewConfig (0.01s)
    --- PASS: TestNewConfig/NewWithoutConfigFileFails (0.01s)
=== RUN   TestNewConfigWithAppRoot
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks_(broker_URLs_override)
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks_(TLS_config_override)
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks_(SASL_config_override,_error)
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks_(SASL_config_override,_error)#01
--- PASS: TestNewConfigWithAppRoot (0.02s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks (0.00s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks_(broker_URLs_override) (0.00s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks_(TLS_config_override) (0.01s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks_(SASL_config_override,_error) (0.00s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks_(SASL_config_override,_error)#01 (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/pubsub     0.058s
=== RUN   TestNewConfig
=== RUN   TestNewConfig/NewWithoutConfigFileFails
--- PASS: TestNewConfig (0.01s)
    --- PASS: TestNewConfig/NewWithoutConfigFileFails (0.01s)
=== RUN   TestNewConfigWithAppRoot
=== RUN   TestNewConfigWithAppRoot/NewWithConfigFileWorks
--- PASS: TestNewConfigWithAppRoot (0.00s)
    --- PASS: TestNewConfigWithAppRoot/NewWithConfigFileWorks (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/server     0.033s
?       github.com/scribd/go-sdk/pkg/testing    [no test files]
?       github.com/scribd/go-sdk/pkg/testing/testproto  [no test files]
=== RUN   TestNewConfig
=== RUN   TestNewConfig/RunningInTestEnvironment
=== RUN   TestNewConfig/NewWithoutConfigFileFails
=== RUN   TestNewConfig/NewWithConfigFile
--- PASS: TestNewConfig (0.03s)
    --- PASS: TestNewConfig/RunningInTestEnvironment (0.00s)
    --- PASS: TestNewConfig/NewWithoutConfigFileFails (0.02s)
    --- PASS: TestNewConfig/NewWithConfigFile (0.00s)
=== RUN   TestSentryHookLevels
--- PASS: TestSentryHookLevels (0.39s)
=== RUN   TestSentryHookDsn
--- PASS: TestSentryHookDsn (0.23s)
=== RUN   TestSentryHookErrorOnInvalidDsn
--- PASS: TestSentryHookErrorOnInvalidDsn (0.18s)
=== RUN   TestSentryHookManualTag
--- PASS: TestSentryHookManualTag (0.13s)
PASS
ok      github.com/scribd/go-sdk/pkg/tracking   0.962s
=== RUN   TestVersion
--- PASS: TestVersion (0.00s)
PASS
ok      github.com/scribd/go-sdk/pkg/version    0.039s
FAIL
Error: running "go test ./... -count=1 -v" failed with exit code 1
ERROR: 1
```

Reply:
I was mistaken and thought it is the warning that is failing the pipeline (oddly enough, I never got the error after adding environments vars). you are right and the problem is caused because of parallel test cases creating same file.
I added new changes, must be fix.

Reply:
@laynax Cool 👍  Then we can remove `CGO_CFLAGS` changes here

Reply:
@laynax Because the warning message itself is not the cause of the issue, the issue was with the tests running in parallel and initializing the DB tables.

As I [mentioned before](https://github.com/scribd/go-sdk/pull/45#discussion_r909545331), this is not a portable change and with other than `g++` C compiler we can get an error. For example, on Mac OS the C compiler is set to `clang` so when I run `mage` I get the following error:

```
error: unknown warning option '-Wno-return-local-addr'; did you mean '-Wno-return-stack-address'? [-Werror,-Wunknown-warning-option]
```

I think if we got the `warning` output, instead of suppressing it we should first understand why it is happening. It is better to be annoyed by the message instead of `hiding` it.

Reply:
@Neurostep I think I understood the problem and there were no need for repeated explanations. I made a change and I was asked to remove it, without asking me why it's there, without telling me why I should delete it.

I understand you're having errors on your machine but the point is to have tests run by docker-compose as suggested in readme.

Now having these as args is actually an out of the box fix for us, just like what [others](https://github.com/scribd/go-sdk/pull/45) did. after adding these, I was going to add new tickets to upgrade gorm to proper version(google did direct update without facing blockage) so it wouldn't be hiding.

having all that said, this thread getting frustrating and I don't think we're getting closer to a solution, I'm removing args.